### ISSUE TITLE
AnalysisMode and AnalysisLevel are not properly respected in .NET 6 RC 1

### ISSUE THREAD

**Issue by DavidZidar** [dotnet/roslyn-analyzers#5529]
### Analyzer source

**SDK**: Built in analyzers in .NET 6 RC 1 SDK.

**Version**: 6.0.100-rc.1.21463.6

### Describe the bug

Even though I set both the following properties...

```xml
<AnalysisMode>AllEnabledByDefault</AnalysisMode>
<AnalysisLevel>latest</AnalysisLevel>
```

... nearly all rules are still disabled.

This is because all of the conditions in `Microsoft.CodeAnalysis.NetAnalyzers.targets` depend on additional properties, for example either `AnalysisLevelDesign` or `AnalysisModeDesign` have to be non empty strings for the design rules to be enabled but they are never set in the build.

Example excerpt:
```xml
Condition="'$(SkipGlobalAnalyzerConfigForPackage)' != 'true' and ('$(AnalysisLevelDesign)' != '' or '$(AnalysisModeDesign)' != '')"
```

### Expected behavior

Most analyzers should be enabled.

### Actual behavior

Nearly all analyzers are disabled.

## Additional context

If I manually add the following lines to my properties than everything works as expected.

```xml
<AnalysisModeDesign>$(AnalysisMode)</AnalysisModeDesign>
<AnalysisModeDocumentation>$(AnalysisMode)</AnalysisModeDocumentation>
<AnalysisModeGlobalization>$(AnalysisMode)</AnalysisModeGlobalization>
<AnalysisModeInteroperability>$(AnalysisMode)</AnalysisModeInteroperability>
<AnalysisModeMaintainability>$(AnalysisMode)</AnalysisModeMaintainability>
<AnalysisModeNaming>$(AnalysisMode)</AnalysisModeNaming>
<AnalysisModePerformance>$(AnalysisMode)</AnalysisModePerformance>
<AnalysisModePublish>$(AnalysisMode)</AnalysisModePublish>
<AnalysisModeReliability>$(AnalysisMode)</AnalysisModeReliability>
<AnalysisModeSecurity>$(AnalysisMode)</AnalysisModeSecurity>
<AnalysisModeUsage>$(AnalysisMode)</AnalysisModeUsage>
```

I have reported this behavior before in #5388 but it was promptly ignored because it was said it was by design, I do not believe this to be the case. This time I am experiencing the same thing in a .NET 6 project.

This is clearly a bug.
